### PR TITLE
Add `health_handler_class` to make `/health` handler is customisable

### PR DESCRIPTION
**Pull request by manics**
The original `/health` handler assumes everything is on Kubernetes, now it's configurable. The default behaviour is unchanged except for an additional field `"_ignored": true`:
```json
{
  "ok": true,
  "checks": [
    {
      "service": "Docker registry",
      "ok": true
    },
    {
      "service": "JupyterHub API",
      "ok": true
    },
    {
      "service": "Pod quota",
      "total_pods": 0,
      "build_pods": 0,
      "user_pods": 0,
      "quota": null,
      "ok": true,
      "_ignored": true
    }
  ]
}
```
I could filter it out, but I think it's worth keeping since it makes it clear that the status of the check is always ignored:
https://github.com/jupyterhub/binderhub/blob/a6f2701196427d11e44415f62dce26870e6cf511/binderhub/health.py#L201-L205

### CODE REVIEW

File: binderhub/health.py [hunk 1]
```diff
@@ -155,58 +127,43 @@ async def check_docker_registry(self):
...
-        # quota doesn't mean the service is unhealthy
-        overall = all(
-            check["ok"] for check in checks if check["service"] != "Pod quota"
-        )
+        # Some checks are for information but do not count as a health failure
+        overall = all(r["ok"] for r in results if not r.get("_ignored", False))
         if not overall:
-            unhealthy = [check for check in checks if not check["ok"]]
+            unhealthy = [r for r in results if not r["ok"]]
             app_log.warning(f"Unhealthy services: {unhealthy}")
```
Comment (consideRatio) [inline binderhub/health.py:165]: Here we will warn about the `_ignored` services as well and not hint that they are accepted to fail. But hmmm, we have the `_ignored` field as part of the `r` dictionary, so maybe its fine as the reader could deduce that.

I think about the naming `_ignore` though, maybe we can make it more descriptive? Its a "check" in a "health check" context, and we don't consider the health to be bad / not ok if this check is failing. What about `_accept_failure` or something like that? We can go with `_ignore` as well if this doesn't resonate with you!

Comment (consideRatio) [reply binderhub/health.py:165]:
```python
# Some checks are for information but do not count as a health failure
```

Maybe even `_count_as_health_failure`?

**Review by consideRatio — APPROVED** [review]
This LGTM and I just commented about the name of the `_ignore` field that I consider if we should name more verbosively, such as `_count_as_health_failure`. We can go with `_ignore` as well but I've grown appreciative of the more descriptive names.

File: binderhub/health.py [hunk 2]
```diff
@@ -218,3 +175,60 @@ async def head(self):
...
+        total_pods = n_user_pods + n_build_pods
+        usage = {
+            "total_pods": total_pods,
+            "build_pods": n_build_pods,
+            "user_pods": n_user_pods,
+            "quota": quota,
+            "ok": total_pods <= quota if quota is not None else True,
+            # The pod quota is treated as a soft quota
+            # Being above quota doesn't mean the service is unhealthy
+            "_ignored": True,
```
Comment (manics) [inline binderhub/health.py:232]:
```suggestion
            "_ignore_failure": True,
```

File: binderhub/health.py [hunk 3]
```diff
@@ -155,58 +127,43 @@ async def check_docker_registry(self):
...
-                check.update(result)
+                results.append(dict({"service": service}, **result))
 
-        # The pod quota is treated as a soft quota this means being above
-        # quota doesn't mean the service is unhealthy
-        overall = all(
-            check["ok"] for check in checks if check["service"] != "Pod quota"
-        )
+        # Some checks are for information but do not count as a health failure
+        overall = all(r["ok"] for r in results if not r.get("_ignored", False))
```
Comment (manics) [inline binderhub/health.py:162]:
```suggestion
        overall = all(r["ok"] for r in results if not r.get("_ignore_failure", False))
```

**Review by manics — COMMENTED** [review]
How about `_ignore_failure`? It's a bit shorter and describes the issue.

Ideally I think we shouldn't return `"ok": false` in the first place, since it's misleading and it clearly is OK!

**Comment by consideRatio** [comment]
Nice `_ignore_failure` is great!

**Pull request merged**